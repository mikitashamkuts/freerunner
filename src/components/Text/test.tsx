/**
 * @format
 */

import 'react-native';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
// Note: import explicitly to use the types shipped with jest.
import {it} from '@jest/globals';

import {Text} from '.';

describe('Text Component', () => {
  it('Renders with default props and matches previous snapshot: ', () => {
    const tree = renderer.create(<Text text="Hello, World!" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders with custom props and matches previous snapshot: ', () => {
    const customProps = {
      text: 'Custom Text',
      fontWeight: 'Bold',
      fontSize: 'Large',
      color: 'Primary',
      numberOfLines: 1,
      accessibilityLabel: 'Custom Label',
      accessibilityHint: 'Custom Hint',
      accessibilityRole: 'header',
      containerStyle: {margin: 10},
    };
    const tree = renderer.create(<Text {...customProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
