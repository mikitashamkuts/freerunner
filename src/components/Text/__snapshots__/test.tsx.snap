// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component Renders with custom props and matches previous snapshot:  1`] = `
<Text
  accessibilityHint="Custom Hint"
  accessibilityLabel="Custom Label"
  accessibilityRole="header"
  numberOfLines={1}
  style={
    [
      {
        "color": undefined,
        "fontFamily": "TestFoundersGrotesk-undefined",
        "fontSize": undefined,
      },
      {
        "margin": 10,
      },
    ]
  }
>
  Custom Text
</Text>
`;

exports[`Text Component Renders with default props and matches previous snapshot:  1`] = `
<Text
  accessibilityRole="text"
  style={
    [
      {
        "color": "#1C1C1C",
        "fontFamily": "TestFoundersGrotesk-Regular",
        "fontSize": 18,
      },
      undefined,
    ]
  }
>
  Hello, World!
</Text>
`;
